### DISCUSSION TITLE
Middleware isn't being triggered

### DISCUSSION THREAD

**Discussion by francistogram** [vercel/next.js#46313] [Help]
Right now my middleware isn't being triggered locally or on production

Next version: `^13.1.5`

I've stripped down `middlware.ts` such that it looks like

```typescript
import { NextResponse } from "next/server";
import type { NextRequest } from "next/server";

export function middleware(request: NextRequest) {
  console.log("IN MIDDLEWARE");
  return NextResponse.next();
}
```

And I don't see anything printed to the server 

I know `middleware.ts` is supposed to be a sibling to pages and my directory structure is
```
├── README.md
├── apiUtils
├── api_middlewares
├── build-storybook.log
├── certs
├── clients
├── components
├── constants
├── db.dump
├── hooks
├── middleware.ts <------- here
├── netlify
├── netlify.toml
├── next-env.d.ts
├── next.config.js
├── package.json
├── pages <------- sibling
├── postcss.config.js
├── prisma
├── providers
├── public
├── server
├── start_scripts
├── stories
├── storybook-static
├── styles
├── supabase
├── tailwind.config.js
├── tsconfig.json
├── types
├── utils
├── yarn-error.log
└── yarn.lock
```

I've checked `tsconfig.json` and made sure to add it to `compilerOptions.include` and not sure where else to start trying to debug if anyone has any pointers

**Comment by francistogram** [comment]
I do see that Middleware does get compiled on `next build` so not sure why it wouldn't be triggered on every page load

<img width="1010" alt="image" src="https://user-images.githubusercontent.com/3721449/220999879-547183ea-ae34-4d4f-abed-64f9c8b3b646.png">

**Comment by icyJoseph** [comment]
Are you using a custom server? 

There's a bunch of tests in the next.js repository that back up this feature, so in order to help further, we need either the repository or a reproduction repository, so we can see what's going on.

**Reply by francistogram** [reply]
Oh I am using a custom server! I set one up so I could get it running on https that was needed for some APIs. Sounds like that may be a good starting point 

I have two files `createSSLCerts.js`
```javascript
const devcert = require("devcert");
const fs = require("fs");

const createSSLCerts = () => {
  if (!fs.existsSync("./certs")) {
    fs.mkdirSync("./certs");
  }

  const domains = ["assembly.local"];

  devcert
    .certificateFor(domains, { getCaPath: true })
    // @ts-ignore
    .then(({ key, cert, caPath }) => {
      fs.writeFileSync("./certs/devcert.key", key);
      fs.writeFileSync("./certs/devcert.cert", cert);
      fs.writeFileSync("./certs/.capath", caPath);
    })
    .catch(console.error);
};

createSSLCerts();
```

and `createLocalServer.js`

```javascript
const { createServer: createHttpsServer } = require("https");
const next = require("next");
const fs = require("fs");

const createLocalServer = () => {
  const dev = process.env.NODE_ENV !== "production";
  const hostname = "assembly.local";
  const port = process.env.NEXT_PUBLIC_PORT || 3000;
  const app = next({ dev, hostname, port });
  const handle = app.getRequestHandler();

  if (!fs.existsSync("./certs/.capath")) {
    const macOsCommand = "sudo yarn ssl:setup";
    const linuxCommand = "yarn ssl:setup";

    console.error("\nError: Missing SSL certificates\n");

    console.error(`To fix this error, run the command below:`);
    console.error(`→ MacOS: ${macOsCommand}`);
    console.error(`→ Linux: ${linuxCommand}\n`);

    process.exit();
  }

  app
    .prepare()
    .then(() => {
      const server = createHttpsServer(
        {
          key: fs.readFileSync("./certs/devcert.key"),
          cert: fs.readFileSync("./certs/devcert.cert"),
        },
        (req, res) => handle(req, res)
      );

      return server.listen(port, (err) => {
        if (err) throw err;

        console.log(`event - ready on https://assembly.local:${port}`);
      });
    })
    .catch((err) => {
      console.error(err);
    });
};

createLocalServer();
```

Anywhere you could point me to so I could get middleware to run on a custom server? I took a look at the [documentations](https://nextjs.org/docs/advanced-features/custom-server) and not sure I see anything

Appreciate the help!

**Reply by icyJoseph** [reply]
Uff yeah I don't see anything wrong either.. what if you take their example for a moment? https://nextjs.org/docs/advanced-features/custom-server

Does it work with their example custom server? Then we know somewhere in between theirs and yours, something is not adding up.

**Reply by francistogram** [reply]
So I went one step further and just removed the custom server altogether

Here's the before and after just so we're on the same page
```json
 "scripts": {
    "dev": "NODE_EXTRA_CA_CERTS=\"$(cat ./certs/.capath)\" node start_scripts/createLocalServer.js",
  }

 "scripts": {
    "dev": "next dev",
  },
```

`middleware.ts` is still just 
```typescript
export async function middleware(req: NextRequest) {
  console.log("in middleware");
  return NextResponse.next();
```

and still don't see anything show up in the server logs printed from the middleware

Have any guidance on how to even start debugging something like this so I can come back with some more information since I realize it's hard (or close to impossible) to help me without being able to replicate on your side

**Reply by francistogram** [reply]
One other thing to note is that it doesn't work doing `next start` either so not a dev issue

**Reply by icyJoseph** [reply]
I'm not sure either. 

If you make another project with create-next-app and add Middleware, does it "work"? 

I wanna discard a few environment things.

Have you maybe disabled logging server side? Maybe some issue with the edge runtime/build? Have you got custom webpack config?

**Reply by francistogram** [reply]
I wish I could tell you I know what fixed it... but randomly started working
<img width="720" alt="image" src="https://user-images.githubusercontent.com/3721449/221285191-d9de99fc-c813-4692-a0ba-e1ca2f947455.png">

Thanks for the help!
